### ISSUE TITLE
SSOi SAML certificate updates

### ISSUE THREAD

**Issue by 7hunderbird** [department-of-veterans-affairs/va.gov-cms#18739]
## User Story or Problem Statement

@timcosgrove brought up an email thread about an SSOi SAML certificate update.

> I am reaching out to you again regarding your SSOi SAML certificate integrated with IAM, expiring on 2024-07-30, next Tuesday. We have not heard back from you about scheduling this change.
To facilitate this change, IAM will need:
Requirements for Update:
Renewed Certificate
In .TXT format or pasted directly into Email response.
Someone to perform the change on the application side.
Someone to test login following the change.
IAM Availability:
Monday – Friday 5PM EST
Thursday 9PM EST *Limited Availability
IAM will be required to submit a Change Request (C.R.), which can take up to two weeks to be approved

When looking at the certificate chain, I could see our `cms.va.gov` is about to expire.

See [comment below](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18739#issuecomment-2249014772) for reference.

The first in the chain at `cms.va.gov` does expire soon.

```
v:NotBefore: Jul  7 23:32:41 2023 GMT; NotAfter: Jul 30 23:32:41 2024 GMT
```

## Description or Additional Context 

Edmund mentioned that it appears our certs are good until 2026, but those are for the *.iam.va.gov certificates and the `cms.va.gov` is due to expire in less than a week.

![CleanShot 2024-07-24 at 16 21 59](https://github.com/user-attachments/assets/b1163c79-2f7e-439a-bee1-2c2d4f6c68a4)

## Steps for Implementation

Create a copy of the https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/cms/coordinate-cert-update-with-iam-team.md file in the `va.gov-team-sensitive` repo and perform similar steps.

## Acceptance Criteria
- [x] SSOi SAML certificate is renewed without disruption to editors.

**Comment by 7hunderbird** [comment]
```
proxychains4 openssl s_client -connect prod.cms.va.gov:443
[proxychains] config file found: /opt/homebrew/etc/proxychains.conf
[proxychains] preloading /opt/homebrew/Cellar/proxychains-ng/4.17/lib/libproxychains4.dylib
[proxychains] DLL init: proxychains-ng 4.17
[proxychains] Strict chain  ...  127.0.0.1:2001  ...  prod.cms.va.gov:443  ...  OK
Connecting to 224.0.0.1
CONNECTED(00000007)
depth=2 DC=gov, DC=va, CN=VA-Internal-S2-RCA2
verify return:1
depth=1 DC=gov, DC=va, CN=VA-Internal-S2-ICA11
verify return:1
depth=0 C=US, ST=D.C., L=Washington, O=U.S. Department of Veterans Affairs, CN=cms.va.gov
verify return:1
---
Certificate chain
 0 s:C=US, ST=D.C., L=Washington, O=U.S. Department of Veterans Affairs, CN=cms.va.gov
   i:DC=gov, DC=va, CN=VA-Internal-S2-ICA11
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Jul  7 23:32:41 2023 GMT; NotAfter: Jul 30 23:32:41 2024 GMT
 1 s:DC=gov, DC=va, CN=VA-Internal-S2-ICA11
   i:DC=gov, DC=va, CN=VA-Internal-S2-RCA2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Nov  7 15:40:12 2022 GMT; NotAfter: Nov  4 15:40:12 2032 GMT
 2 s:DC=gov, DC=va, CN=VA-Internal-S2-RCA2
   i:DC=gov, DC=va, CN=VA-Internal-S2-RCA2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Oct 12 18:28:08 2022 GMT; NotAfter: Oct 12 18:37:51 2042 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIFfDCCBGSgAwIBAgIHPQAAAAA+pTANBgkqhkiG9w0BAQsFADBIMRMwEQYKCZIm
iZPyLGQBGRYDZ292MRIwEAYKCZImiZPyLGQBGRYCdmExHTAbBgNVBAMTFFZBLUlu
dGVybmFsLVMyLUlDQTExMB4XDTIzMDcwNzIzMzI0MVoXDTI0MDczMDIzMzI0MVow
dDELMAkGA1UEBhMCVVMxDTALBgNVBAgTBEQuQy4xEzARBgNVBAcTCldhc2hpbmd0
b24xLDAqBgNVBAoTI1UuUy4gRGVwYXJ0bWVudCBvZiBWZXRlcmFucyBBZmZhaXJz
MRMwEQYDVQQDEwpjbXMudmEuZ292MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
CgKCAQEAr/RfoygoIWb5Nok9WsMHKZJ37/7+UHhES3QuIq2JAREZG+ZfuGJOUkTm
oVOY7sxe3jWI6/VtPEyZs31J7Q+j660UbKl0rWfFyLxY7GmFS4OfLp1hcXENLCmU
yjQXIYzidCyghjMl8tPcPpV/5xrz89s9KbjDpLwdbm0hFhpyOcnJfVlkdZFfWCwN
ETE7FYEcfL2c6aGq/QrdL8JLIUZinqzVdGLbCNanomFP/Xezn00h2uTQt6+lCNCp
J7G1YTZf/XVMAWRjguDuY/N1aaXoizpRXiBHq1QUtM9Poy6tTwmQRk2H8CMRsG4Q
NXAWUk2nEG7aWcO5R7mQcKZUje2RgQIDAQABo4ICPTCCAjkwgZAGA1UdEQSBiDCB
hYIKY21zLnZhLmdvdoIOZGV2LmNtcy52YS5nb3aCEnN0YWdpbmcuY21zLnZhLmdv
doIPcHJvZC5jbXMudmEuZ292ghN0ZXN0LmRldi5jbXMudmEuZ292ghd0ZXN0LnN0
YWdpbmcuY21zLnZhLmdvdoIUdGVzdC5wcm9kLmNtcy52YS5nb3YwHQYDVR0OBBYE
FCjAeLug/5cEQYNkPtupFv1w6gJLMB8GA1UdIwQYMBaAFCJYiBkpF4GWnffJqBgs
dSp8FCypMEcGA1UdHwRAMD4wPKA6oDiGNmh0dHA6Ly9jcmwucGtpLnZhLmdvdi9w
a2kvY3JsL1ZBLUludGVybmFsLVMyLUlDQTExLmNybDB5BggrBgEFBQcBAQRtMGsw
RQYIKwYBBQUHMAKGOWh0dHA6Ly9haWEucGtpLnZhLmdvdi9wa2kvYWlhL3ZhL1ZB
LUludGVybmFsLVMyLUlDQTExLmNlcjAiBggrBgEFBQcwAYYWaHR0cDovL29jc3Au
cGtpLnZhLmdvdjAMBgNVHRMBAf8EAjAAMAsGA1UdDwQEAwIE8DA9BgkrBgEEAYI3
FQcEMDAuBiYrBgEEAYI3FQiByMMzgfnwBoGlnw2E4IEIhcKqSwaFlI4BgqKIPwIB
ZAIBLTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwJwYJKwYBBAGCNxUK
BBowGDAKBggrBgEFBQcDATAKBggrBgEFBQcDAjANBgkqhkiG9w0BAQsFAAOCAQEA
r5zL24b8BIZht2u4B08O5EA/nfx/EyCmYdDINYnPtMhEnufPRzJCUlb7MO3gM5Yo
A7vHAAdBUHx11kX6XX06so0d7hGb7A7/xyYJy6j6NJp7uDrxI1xrJMxCntaEz/O7
dnfpPNrm07xW/hkJCgjQ0AC4stFSmMYoUEwpCklN66KokPoFqlMLL56VrkgK/NXQ
hUYz1HccqZoHrDJ4nvlSO2rUeL2e/OAqCElXnoeu8PgEA3hw6WW/nMKSssHgFySb
Vhmja0o3EE0dajJK8t8Qh57ODttjCeZrPFZjr/v3iZsUym7teUZfri9HQ72sk2WB
HMo4SUz3XYkIJj5vu9tBHA==
-----END CERTIFICATE-----
subject=C=US, ST=D.C., L=Washington, O=U.S. Department of Veterans Affairs, CN=cms.va.gov
issuer=DC=gov, DC=va, CN=VA-Internal-S2-ICA11
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA
Server Temp Key: ECDH, prime256v1, 256 bits
---
SSL handshake has read 4116 bytes and written 449 bytes
Verification: OK
---
New, TLSv1.2, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 28C6FB04B563A15F3418849D19650B75AB5E07919E3F1826835D5ABBAD740752
    Session-ID-ctx:
    Master-Key: F7C28B36BAE1B94F66C4FD1909506FDAFEBC2E639F62E967759BC4E5078D61EDDCC1E2A629EA7224CD49CAA8A6085107
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 86400 (seconds)
    TLS session ticket:
    0000 - 1b f0 02 9d 30 fb d7 27-ad be b8 cf ea 66 2c 73   ....0..'.....f,s
    0010 - 7a 9d 6d da 17 6b 73 f9-d6 7c c2 9b ff 7d 29 0a   z.m..ks..|...}).
    0020 - d4 48 57 e0 07 ec 53 31-6e 04 a7 3e 52 21 8f b2   .HW...S1n..>R!..
    0030 - b1 25 b9 a3 3e aa b4 10-44 9d 7b 46 e4 5d ec 3b   .%..>...D.{F.].;
    0040 - da ad d9 82 76 31 4c 4f-fe c7 91 c3 58 fa 75 ef   ....v1LO....X.u.
    0050 - 11 95 af f6 45 3e 29 fe-1e da 75 aa ce 36 b3 73   ....E>)...u..6.s
    0060 - ff a0 f1 1b 2c 8a 42 a7-6f                        ....,.B.o

    Start Time: 1721860713
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: yes
---
```

**Comment by 7hunderbird** [comment]
When I searched for who owns `cms.va.gov` I started my search in VFS confluence.  

The first result I checked was:

- https://vfs.atlassian.net/wiki/spaces/OT/pages/2418507777/Monitoring+and+tracking+the+expiration+of+SSL+TLS+certificates

Which lead me to look into who owns `cms.va.gov` in the `va.gov-team-sensitive` repo.

I found this article:

https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/cms/cms-sso-saml-iam-login-braindump.md

**Comment by 7hunderbird** [comment]
We got a certificate with @ph-One's help.  With this certificate we were able to rehearse and test deploying the certificate and create a sequence of steps.

- [Procedure](https://vfs.atlassian.net/wiki/spaces/~712020436a1be296514af5b58f0846a4d9d910/pages/3327295549/Procedure+to+Renew+cms.va.gov+Cert+on+Staging+and+Prod)

We worked with members of the IAM team like jonathan.outlaw@va.gov and over a Microsoft Teams meeting we ran pre-deploy smoke tests, performed the deploy, post-deploy smoke tests and can confirm that the cert is updated and PIV card access works as intended.
